identity/src/main/java/com/android/identity/trustmanagement/TrustManager.kt

```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright 2023 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.android.identity.trustmanagement
+
+import java.security.cert.CertificateException
+import java.security.cert.PKIXCertPathChecker
+import java.security.cert.X509Certificate
+
+/**
+ * This class is used for the verification of a certificate
+ * chain.
+ *
+ * The user of the class can add trust roots using method
+ * [addCertificate]. At this moment certificates of type
+ * [X509Certificate] are supported.
+ *
+ * The Subject Key Identifier (extension 2.5.29.14 in the
+ * [X509Certificate]) is used as the primary key / unique
+ * identifier of the root CA certificate. In the verification
+ * of the chain this will be matched with the Authority Key
+ * Identifier (extension 2.5.29.35) of the certificate issued
+ * by this root CA.
+ */
```

Comment:
What happens if I use `TrustManager.verify()` with a certificate chain where not all the certificates have _X509v3 Authority Key Identifier_ or _X509v3 Subject Key Identifier_ set?

Reply:
Certificates without the Extension will be skipped, but this will not result in a technical error. We suppose the certificate chain in the document will at least have one certificate with the extension in it.